File: dbc_talks/caeser_cipher/caeser_cipher.js
var encodeChar = function(char, shift) {
  if (/[A-Za-z]/.test(char)) {
    if (/[A-Z]/.test(char)) {
      var offset = (ord(char) - ord("A") + shift) % 26;
      return chr(ord("A") + offset)
    }
    else {
      var offset = (ord(char) - ord("a") + shift) % 26;
      return chr(ord('a') + offset)
    }
  }
  else { return char }
}

var ord = function(char) {
  return char.charCodeAt(0);
}

var chr = function(int) {
  return String.fromCharCode(int);
}






console.log(encodeChar("A", 3) === "D")
console.log(encodeChar("a", 4) === "e")
console.log(encodeChar("Z", 2) === "B")
console.log(encodeChar("z", 1) === "a")

File: dbc_talks/caeser_cipher/caeser_cipher.rb
def encode_char(char, shift)
  if /[A-Za-z]/ =~ char
    if /[A-Z]/ =~ char
      offset = (char.ord - "A".ord + shift) % 26
      ("A".ord + offset).chr
    else
      offset = (char.ord - "a".ord + shift) % 26
      ("a".ord + offset).chr
    end
  else
    char
  end
end

def encode_string(string, shift)
  string.split('').map{|char| encode_char(char, shift)}.join('')
end

def decode_string(string, shift)
  encode_string(string, -shift)
end


puts encode_char("A", 3) == "D"
puts encode_char("b", 4) == "f"
puts encode_char("Z", 3) == "C"
puts encode_char("z", 2) == "b"
puts encode_char(" ", 2) == " "
puts encode_char(",", 2) == ","

puts encode_string('Abcdef', 3) == 'Defghi'
puts decode_string(encode_string('Abcdef', 3), 3) == 'Abcdef'

